### PR TITLE
testing/solidity: new aport responding to #10660

### PR DESCRIPTION
**Pull request by b17wise**
URL: https://solidity.readthedocs.io/en/latest/index.html

Description: 
Please look into the issue: https://bugs.alpinelinux.org/issues/10660?next_issue_id=10659

### CODE REVIEW

**Comment by maxice8** [comment]
Use our commit naming scheme

`<repo>/<package>: new aport`

$url
$pkgdesc

should follow in the **body** of the commit

File: testing/solidity/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Contributor: Trevis Schiffer <nikolaibitinit@gmail.com>
+# Maintainer: Trevis Schiffer <nikolaibitinit@gmail.com>
+
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: unnecessary whitespace

File: testing/solidity/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+pkgname=solidity
+pkgver=0.5.10
+pkgrel=1
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: pkgrel starts at 0

File: testing/solidity/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+pkgdesc="The Solidity Contract-Oriented Programming Language"
+url="https://github.com/ethereum/solidity"
+arch="all"
+license="GPL3"
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: GPL3 is not a SPDX license identifier

File: testing/solidity/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+depends=""
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: empty variables can be removed for brevity

File: testing/solidity/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+makedepends="boost-dev cmake git"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/ethereum/$pkgname/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
+options="!check" # Test suite issues
+
+prepare() {
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: missing default_prepare on top

File: testing/solidity/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+license="GPL3"
+depends=""
+makedepends="boost-dev cmake git"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/ethereum/$pkgname/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
+options="!check" # Test suite issues
+
+prepare() { 
+	cd $builddir
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: cd $builddir can be removed

File: testing/solidity/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	git init
+	echo -n "" > commit_hash.txt
+	# Fix for GCC 8 
+	sed -e '/add_compile_options(-Wall)/d' -i.bak cmake/EthCompilerSettings.cmake
+}
+
+build() { 
+	cd "$builddir" 
+	mkdir -p build 
+	cd build
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: creating the build directory for out-of-tree builds usually happens in prepare()

File: testing/solidity/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+prepare() { 
+	cd $builddir 
+	git init
+	echo -n "" > commit_hash.txt
+	# Fix for GCC 8 
+	sed -e '/add_compile_options(-Wall)/d' -i.bak cmake/EthCompilerSettings.cmake
+}
+
+build() { 
+	cd "$builddir"
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: cd "$builddir" can be removed

File: testing/solidity/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+build() { 
+	cd "$builddir" 
+	mkdir -p build 
+	cd build 
+	cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBoost_USE_STATIC_LIBS=OFF -DTESTS=OFF -DCMAKE_BUILD_TYPE=Release
+	make -i
+}
+
+package() { 
+	cd "$builddir"
```
Comment (maxice8) [inline testing/solidity/APKBUILD]: cd "$builddir" can be removed

**Comment by b17wise** [comment]
Awesome...!
Will fix these issues

File: testing/solidity/APKBUILD (hunk 5 above)
Comment (b17wise) [reply testing/solidity/APKBUILD]: Reference says it only requires for applying patches..
What I am missing in this case?

Comment (maxice8) [reply testing/solidity/APKBUILD]: It is expected to leave there so people aren't surprised why their patches aren't applied

Comment (b17wise) [reply testing/solidity/APKBUILD]: I see

File: testing/solidity/APKBUILD [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+license="GPL-3.0-or-later"
+depends="libgcc"
+makedepends="boost-dev boost cmake git"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/ethereum/$pkgname/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
+options="!check" # Test suite issues
+
+default_prepare() {
+}
```
Comment (Cogitri) [inline testing/solidity/APKBUILD]: You're supposed to call `default_prepare` in `prepare`

File: testing/solidity/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+pkgrel=0
+pkgdesc="The Solidity Contract-Oriented Programming Language"
+url="https://github.com/ethereum/solidity"
+arch="all"
+license="GPL-3.0-or-later"
+depends="libgcc"
+makedepends="boost-dev boost cmake git"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/ethereum/$pkgname/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
```
Comment (Cogitri) [inline testing/solidity/APKBUILD]: Default value, can be removed

File: testing/solidity/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+options="!check" # Test suite issues
```
Comment (Cogitri) [inline testing/solidity/APKBUILD]: Please be more precise, what's actually going wrong?

File: testing/solidity/APKBUILD [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+default_prepare() {
+}
+
+prepare() { 
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/solidity/APKBUILD]: As maxice noted, this is unnecessary

File: testing/solidity/APKBUILD [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+	cd "$builddir" 
+	mkdir -p build 
+	echo -n "" > commit_hash.txt
+	# Fix for GCC 8 
+	sed -e '/add_compile_options(-Wall)/d' -i.bak cmake/EthCompilerSettings.cmake
+}
+
+build() { 
+	cd build 
+	cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBoost_USE_STATIC_LIBS=OFF -DTESTS=OFF -DCMAKE_BUILD_TYPE=Release
```
Comment (Cogitri) [inline testing/solidity/APKBUILD]: Please split this into multiple lines, it's over 80 chars long

File: testing/solidity/APKBUILD (hunk 11 above)
Comment (b17wise) [reply testing/solidity/APKBUILD]: This one?  builddir="$srcdir/$pkgname-$pkgver"

Comment (Cogitri) [reply testing/solidity/APKBUILD]: Yes, that's the default value, so you can remove the builddir.

File: testing/solidity/APKBUILD (hunk 12 above)
Comment (b17wise) [reply testing/solidity/APKBUILD]: test added

**Comment by maxice8** [comment]
Updated with a fix to the commits, check it to see how they are supposed to be done.
